packages/hub/tasks/scheduled-payment-on-chain-cancelation-waiter.ts

```diff
@@ -0,0 +1,99 @@
+import { inject } from '@cardstack/di';
+import * as Sentry from '@sentry/node';
+import { isBefore, subDays } from 'date-fns';
+import { nowUtc } from '../utils/dates';
+
+export default class ScheduledPaymentOnChainCancelationWaiter {
+  prismaManager = inject('prisma-manager', { as: 'prismaManager' });
+  cardpay = inject('cardpay');
+  ethersProvider = inject('ethers-provider', { as: 'ethersProvider' });
+
+  async perform(payload: { scheduledPaymentId: string }) {
+    let prisma = await this.prismaManager.getClient();
+    let scheduledPayment = await prisma.scheduledPayment.findFirstOrThrow({
+      where: { id: payload.scheduledPaymentId },
+    });
+
+    let provider = this.ethersProvider.getInstance(scheduledPayment.chainId);
+    let scheduledPaymentModule = await this.cardpay.getSDK('ScheduledPaymentModule', provider);
+
+    if (scheduledPayment.cancelationBlockNumber) {
+      // Did we somehow spawn this task after it already updated cancelationBlockNumber previously?
+      return Sentry.captureException(
+        new Error('Task run for scheduled payment that already has a cancelation block number'),
+        {
+          tags: {
+            action: 'scheduled-payment-task',
+            scheduledPaymentId: scheduledPayment.id,
+          },
+        }
+      );
+    }
```

Comment:
I saw assertions for some of these outcomes but not others, including this, is it worth having these exercised in tests?

Reply:
Thanks - added a couple of test cases.